src/CommunityToolkit/Xamarin.CommunityToolkit/Views/Popup/iOS/PopupRenderer.ios.cs

```diff
@@ -140,12 +140,12 @@ void CreateControl()
 		void SetViewController()
 		{
 			IVisualElementRenderer currentPageRenderer;
-			var modalStackCount = Application.Current.MainPage?.Navigation?.ModalStack?.Count ?? 0;
 			var page = Application.Current.MainPage;
+			var modalStackCount = page?.Navigation?.ModalStack?.Count ?? 0;
 			if (modalStackCount > 0)
 			{
 				var index = modalStackCount - 1;
-				page = page!.Navigation!.ModalStack![index];
+				page = page?.Navigation.ModalStack[index];
```

Comment:
if `ModalStack` is > 0 `Page` will not be null

Reply:
I know, but we receive an error if remove null propagation. and we discussed not to use Null forgiving operator
![image](https://user-images.githubusercontent.com/33021114/146812302-99c01132-dee5-47fd-bd99-7ce14cf2c7b4.png)

Reply:
rewrite it and add check page for null